src/native/minipal/thread.h

```diff
@@ -43,9 +43,6 @@ extern "C" {
  */
 static inline size_t minipal_get_current_thread_id(void)
 {
-#if defined(__wasm) && !defined(_REENTRANT)
```

Comment:
Given `!defined(_REENTRANT)`, I assume that this is an optimization for single-threaded wasm. There is one-and-only thread ID for single-threaded wasm and thus the function can return a constant.

Are we ok with losing this optimization? Is it going to regress perf in something important?

If `0` does not work well, we can change it to return `1` and keep the optimization.

Reply:
> _REENTRANT is inconsistent in practice, but perhaps things have changed since I last hiot confusion

`_REENTRANT` is a reliable indicator of "multi-threaded" vs "single-threaded" on WASM. It is set by clang and is used in all the standard libraries and such.

If I were to changing this to `(size_t)(void*)pthread_self`, I'd be more worried about the diagnostics experience degradation, since it is going to be a random pointer into the data section (without threads, I haven't checked what the threading impl has it as).

Reply:
Okay. Thanks for the context @SingleAccretion. I've updated this to return a non-zero value.